orchestrator-bundle/orc8r-bootstrapper-operator/src/charm.py

```diff
@@ -107,42 +100,97 @@ def _service_is_running(self) -> bool:
         return False
 
     @property
-    def _certs_are_stored(self) -> bool:
-        """Returns whether the bootstrapper private key is stored.
+    def _bootstrapper_private_key(self) -> Optional[str]:
+        """Returns bootstrapper private key.
 
         Returns:
-            bool: True/False
+            str: Private key
         """
-        return self._container.exists(f"{self.CERTIFICATE_DIRECTORY}/bootstrapper.key")
+        replicas = self.model.get_relation("replicas")
+        if not replicas:
+            return None
+        return replicas.data[self.app].get("bootstrapper_private_key", None)
 
     @property
-    def _cert_bootstrapper_relation_created(self) -> bool:
-        """Returns whether cert-bootstrapper relation is created.
+    def _bootstrapper_private_key_is_stored(self) -> bool:
+        """Returns whether certificates are stored in relation data.
```

Comment:
it seems to me as it checks if the `private key` is stored not `certificates`

```suggestion
        """Returns whether bootstrapper private key is stored in relation data.
```